File: 26.remove-duplicates-from-sorted-array.py
#
# @lc app=leetcode id=26 lang=python3
#
# [26] Remove Duplicates from Sorted Array
#

# @lc code=start
class Solution:
    def removeDuplicates(self, nums: List[int]):
        i = 0
        j = 1
        n = len(nums)-1
        while n>0:
            if nums[i] == nums[j]:
                del nums[i]
            else:
                i+=1
                j+=1
            n-=1

if __name__ == '__main__':
    a = Solution()
    b = a.removeDuplicates([1,1,2])
    print(b)
# @lc code=end

File: C#/26.remove-duplicates-from-sorted-array.cs
/*
 * @lc app=leetcode id=26 lang=csharp
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
public class Solution {
    public int RemoveDuplicates(int[] nums) {
        if(nums.Length == 0) return 0;
        int cur = nums[0];
        int cnt = 1;
        for(int i = 0; i < nums.Length; i ++){
            if(nums[i] != cur){
                cur = nums[i];
                nums[cnt] = nums[i];
                cnt ++;
            }
        }
        return cnt;
    }
}
// @lc code=end

